zerver/lib/events.py

```diff
@@ -375,7 +375,8 @@ def fetch_initial_state_data(
             client_gravatar=False,
         )
 
-        state["can_create_streams"] = settings_user.can_create_streams()
+        state["can_create_private_streams"] = settings_user.can_create_private_streams()
+        state["can_create_public_streams"] = settings_user.can_create_public_streams()
```

Comment:
I think we need to preserve the legacy `can_create_streams` boolean, because `zulip-module` currently reads it.  Probably the right model is to have its value be `settings_user.can_create_private_streams() || settings_user.can_create_public_streams()`, and make clear in the documentation that it's deprecated in favor of the new properties.

Reply:
@timabbott We discussed that we should avoid these fields like `can_create_private_streams` here and clients can just compute them using the policy value. We will need to keep `can_create_streams` for legacy reasons as you mentioned.